src/gui/attributetable/qgsdualview.cpp

```diff
@@ -146,11 +146,11 @@ void QgsDualView::init( QgsVectorLayer *layer, QgsMapCanvas *mapCanvas, const Qg
     mAttributeEditor->layout()->addWidget( mAttributeForm );
   }
 
-  connect( mAttributeForm, &QgsAttributeForm::widgetValueChanged, this, &QgsDualView::featureFormAttributeChanged );
-  connect( mAttributeForm, &QgsAttributeForm::modeChanged, this, &QgsDualView::formModeChanged );
-  connect( mMasterModel, &QgsAttributeTableModel::modelChanged, mAttributeForm, &QgsAttributeForm::refreshFeature );
-  connect( mAttributeForm, &QgsAttributeForm::filterExpressionSet, this, &QgsDualView::filterExpressionSet );
-  connect( mFilterModel, &QgsAttributeTableFilterModel::sortColumnChanged, this, &QgsDualView::onSortColumnChanged );
+  connect( mAttributeForm, &QgsAttributeForm::widgetValueChanged, this, &QgsDualView::featureFormAttributeChanged, Qt::UniqueConnection );
```

Comment:
_mAttributeForm_ is deleted just above so the connection should be removed, same for _mMasterModel_ deleted in initModels (maybe the connect should be moved in that method)

Reply:
Hm... I will go over it again. Thanks.

Reply:
I agree that it does not need to be unique since the connection are deleted before anyway. I think we cannot move the connections with the `mMasterModel` to the `initModels` since it connects to the `mAttributeForm` created later. (same for `mFilterModel`).

Reply:
Right